src/main/scala/nl.knaw.dans.easy.solr4files/package.scala

```diff
@@ -60,26 +59,38 @@ package object solr4files extends DebugEnhancedLogging {
 
   implicit class RichURL(val left: URL) {
 
-    def loadXml: Try[Elem] = Try {
-      resource.managed(left.openStream()).acquireAndGet(XML.load) // TODO replace
+    def loadXml: Try[Elem] = {
+      getContent.flatMap(s => Try(XML.loadString(s)))
```

Comment:
since `getContent` is already a `Try`, you don't need to wrap `XML.loadString` in a `Try` anymore. Any kind of error handling will be dealt with by `map`.

`getContent.map(XML.loadString)`